modules/core/src/opengl.cpp

```diff
@@ -45,6 +45,13 @@
 #ifdef HAVE_OPENGL
 #  include "gl_core_3_1.hpp"
 #  ifdef HAVE_CUDA
+#    if defined(__arm__) || defined(__aarch64__)
+#      warning Workaround for CUDA on ARM
```

Comment:
To make robust solution please add option to bypass this workaround:
```
if (defined(__arm__) || defined(__aarch64__)) \
    && !defined(OPENCV_SKIP_CUDA_OPENGL_ARM_WORKAROUND)
```

which can be applied via C++ CMake flags.

Reply:
Should this workaround be enabled or disabled by default? How should it called `FORCE_GL_CUDA_ARM` or `DISABLE_GL_CUDA_ARM` or???
The current change is derived from the patch for CUDAs to make OpenCV compile with it, and covers exactly the cases which CUDA currently rejects:
```diff --git a/cuda_gl_interop.h b/cuda_gl_interop.h
index 0f4aa17..e8c538c 100644
--- a/cuda_gl_interop.h
+++ b/cuda_gl_interop.h
@@ -59,13 +59,13 @@
 
 #else /* __APPLE__ */
 
-#if defined(__arm__) || defined(__aarch64__)
-#ifndef GL_VERSION
-#error Please include the appropriate gl headers before including cuda_gl_interop.h
-#endif
-#else
+//#if defined(__arm__) || defined(__aarch64__)
+//#ifndef GL_VERSION
+//#error Please include the appropriate gl headers before including cuda_gl_interop.h
+//#endif
+//#else
 #include <GL/gl.h>
-#endif
+//#endif
 
 #endif /* __APPLE__ */
```
re: `patch-1` it is an artifact from github editor, will try to rename them before making PRs

Reply:
Thank you for update!

I don't see information about fixed CUDA versions in linked forum thread, so this workaround should be enabled by default (but probably we should eliminate `#warning` from here or disable that message under another "*_MESSAGE" definition flag).

Please change "default" value of `CL_VERSION` to `0x1F02`. I see this on my machine (Fedora):
```
/usr/include/GL/gl.h:#define GL_VERSION				0x1F02
/usr/include/GL/glcorearb.h:#define GL_VERSION                        0x1F02
```

I believe CMake changes are not necessary (option is very specific, lets keep it hidden).
Flag can be added via CMake for future fixed versions (if necessary - workaround is harmless without warning message and proper default value):
```
cmake ... -DOPENCV_EXTRA_FLAGS=-DOPENCV_SKIP_CUDA_OPENGL_ARM_WORKAROUND=1
```